nocaselist/_nocaselist.py

```diff
@@ -0,0 +1,374 @@
+"""
+The only class exposed by this package is :class:`nocaselist.NocaseList`.
+"""
+
+from __future__ import print_function, absolute_import
+
+__all__ = ['NocaseList']
+
+
+def _lc_list(lst):
+    """
+    Return a lower-cased list from the input list.
+    """
+    result = list()
+    for value in lst:
+        result.append(value.lower())
+    return result
+
+
+class NocaseList(list):
+    # pylint: disable=line-too-long
+    """
+    A case-insensitive, but case-preserving, list.
+
+    The list is case-insensitive: Whenever items of the list are looked up by
+    value or item values are compared, that is done case-insensitively. The
+    case-insensitivity is defined by performing the lookup or comparison at
+    the level of the result of the ``lower()`` method on list items. Therefore,
```

Comment:
suggest remove "at the level of" and replace 'at' with 'on'

Reply:
DONE